src/test/ui/issues/issue-21763.stderr

```diff
@@ -4,10 +4,10 @@ error[E0277]: `std::rc::Rc<()>` cannot be sent between threads safely
 LL |     foo::<HashMap<Rc<()>, Rc<()>>>();
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `std::rc::Rc<()>` cannot be sent between threads safely
    |
-   = help: within `std::collections::HashMap<std::rc::Rc<()>, std::rc::Rc<()>>`, the trait `std::marker::Send` is not implemented for `std::rc::Rc<()>`
+   = help: within `(std::rc::Rc<()>, std::rc::Rc<()>)`, the trait `std::marker::Send` is not implemented for `std::rc::Rc<()>`
    = note: required because it appears within the type `(std::rc::Rc<()>, std::rc::Rc<()>)`
-   = note: required because it appears within the type `std::marker::PhantomData<(std::rc::Rc<()>, std::rc::Rc<()>)>`
-   = note: required because it appears within the type `std::collections::hash::table::RawTable<std::rc::Rc<()>, std::rc::Rc<()>>`
+   = note: required because of the requirements on the impl of `std::marker::Send` for `hashbrown::raw::RawTable<(std::rc::Rc<()>, std::rc::Rc<()>)>`
+   = note: required because it appears within the type `hashbrown::map::HashMap<std::rc::Rc<()>, std::rc::Rc<()>, std::collections::hash_map::RandomState>`
```

Comment:
Do I understand from this UI test update that the compiler may "leak" to the user that `std::collections::HashMap` is actually implemented in the `hashbrown` crate? I think this might be confusion to users.

Reply:
It’s not especially different from the current situation of exposing implementation types like RawTable or BTreeNode

Reply:
Except those are all types from `std` itself. So the user might see a private type name but at least they know where it's coming from.

Reply:
This kind of thing already caused a lot of confusion with the "simple" issue of re-exporting of `core` via `std`. I feel like I haven't seen that problem in a while; is there some existing solution we can use?

Reply:
There is a way to do this reexport that produces a path that starts with `std`, I just don't know how. There are reexports of `core` / `alloc` items in `std` that handle this correctly.

Reply:
std::HashMap isn't a re-export, though. It's a newtype around hashbrown::Hashmap

Reply:
You could also reexport `hashbrown` as an "implementation detail" or something, but I don't know if you can get `rustc` to use `std` paths in error messages without it being publicly reachable, which we'd want to avoid.

Reply:
You can also, for this error in particular, have `unsafe impl<K: Send, V: Send, S: Send> Send for HashMap<K, V, S> {}` but I'm not sure that's a good idea.

Reply:
>  I don't know if you can get rustc to use std paths in error messages without it being publicly reachable, which we'd want to avoid.

What if you re-export them as pub items, but the re-export itself is not actually publicly reachable?